### PR TITLE
Teams integration

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Automated integration into the Teams SDK

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [x] Other

## Summary by CodeRabbit

- **New Features**
  - Introduced a new job to trigger a workflow in another repository with specific configurations.
  - Enhanced the existing test job to include the new workflow job as a dependency.

- **Chores**
  - Updated the release workflow to improve checkout efficiency with a `fetch-depth: 0` parameter.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -19,9 +19,27 @@ jobs:
...
+    runs-on: ubuntu-latest
+    if: github.base_ref == 'develop'
+    steps:
+      - uses: convictional/trigger-workflow-and-wait@v1.6.1
+        with:
+          owner: voxel51
+          repo: fiftyone-teams
+          github_token: ${{ secrets.FIFTYONE_GITHUB_TOKEN }}
+          github_user: voxel51-bot
+          workflow_file_name: merge-oss.yml
```
Comment (ritch) [inline .github/workflows/pr.yml:32]: Can you link to this file?

Comment (benjaminpkane) [reply .github/workflows/pr.yml:32]: Yes, the automation can find this in the private repository. Can take offline

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -10,21 +10,42 @@ on:
       - release/v[0-9]+.[0-9]+.[0-9]+
 
 jobs:
-  lint:
-    uses: ./.github/workflows/lint-app.yml
-
```
Comment (findtopher) [inline .github/workflows/pr.yml:15]: did you alphabetize the jobs?  I... I've never been so happy!

**Review by findtopher — APPROVED** [review]
this is gonna be fun to watch!

**Pull request merged**